<?xml version="1.0" encoding="UTF-8"?>

<launch>
    <!-- Set dynamic_reconfigure parameters -->
    <!--<include file="$(find ghost_bridge)/launch/dynamic_reconfigure_params.launch" />-->

    <!-- RealSense camera settings -->
    <arg name="camera_width" default="1280"/>
    <arg name="camera_height" default="720"/>
    <arg name="camera_fps" default="30"/>

    <include file="$(find realsense2_camera)/launch/rs_aligned_depth.launch">
        <arg name="depth_width" value="$(arg camera_width)"/>
        <arg name="depth_height" value="$(arg camera_height)"/>
        <arg name="enable_depth" value="true"/>
        <arg name="depth_fps" value="$(arg camera_fps)"/>

        <arg name="infra1_width" value="$(arg camera_width)"/>
        <arg name="infra1_height" value="$(arg camera_height)"/>
        <arg name="enable_infra1" value="false"/>

        <arg name="infra2_width" value="$(arg camera_width)"/>
        <arg name="infra2_height" value="$(arg camera_height)"/>
        <arg name="enable_infra2" value="false"/>

        <arg name="color_width" value="$(arg camera_width)"/>
        <arg name="color_height" value="$(arg camera_height)"/>
        <arg name="enable_color" value="true"/>
        <arg name="color_fps" value="$(arg camera_fps)"/>
    </include>

    <!-- Perception -->
    <arg name="perception_fps" default="20"/>

    <include file="$(find ros_people_model)/launch/perception.launch">
        <arg name="face_detector_topic" value="/camera/color/image_raw"/>
        <arg name="perception_fps" value="$(arg perception_fps)"/>
    </include>

    <!-- Models -->
    <include file="$(find ros_people_model)/launch/models.launch">
        <arg name="perception_fps" value="$(arg perception_fps)"/>
    </include>

    <!-- Face tracker -->
    <node name="face_tracker" pkg="ghost_bridge" type="face_tracker_node.py" output="screen"/>

    <!-- Face tracker -->
    <node name="mirroring" pkg="ros_people_model" type="mirroring.py" output="screen"/>
    <node pkg="tf" type="static_transform_publisher" name="audience_broadcaster" args="0 0 0 0 0 0 1 base_link audience 100" />

    <!-- Ghost Bridge -->
    <node name="faces_throttler" type="throttle" pkg="topic_tools" args="messages /faces 0.3 /faces_throttled" output="screen"/>
    <node name="ghost_bridge" pkg="ghost_bridge" type="ghost_bridge_node.py" output="log"/>

  <node name="video_recorder" pkg="video_recorder" type="recorder.py" output="screen" required="true">

      <param name="output_width" type="int" value="960" />
      <param name="output_height" type="int" value="540" />
      <param name="output_format" value="mp42" />
      <param name="output_path" value="/home/hr/VIDEOLOG/camera_[timestamp].avi" />

      <param name="output_topic" value="/video_recorder/image_raw" />

      <param name="source1" value="/camera/color/image_raw,0,0,960,540" />
  </node>


</launch>
